### PR TITLE
Improve assert messages in mmds::token::test_compute_expiry

### PR DESCRIPTION
**Pull request by roypat**
Signed-off-by: Patrick Roy <roypat@amazon.co.uk>

## Changes

In case the assertions fail, print out the values

## Reason

We have observed spurious failures in this test, and this hopefully helps debugging them

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [x] All commits in this PR are signed (`git commit -s`).
- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] New `unsafe` code is documented.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md

### CODE REVIEW

File: src/mmds/src/token.rs [hunk 1]
```diff
@@ -378,12 +378,23 @@ mod tests {
...
+            MILLISECONDS_PER_SECOND
+        );
 
         let time_now = get_time_ms(ClockType::Monotonic);
         let expiry = TokenAuthority::compute_expiry(0);
         let ttl = expiry - time_now;
-        assert!(ttl <= deviation);
+        assert!(
+            ttl <= deviation,
+            "ttl={} is greater than {}",
```
Comment (pb8o) [inline src/mmds/src/token.rs:394]: nit: Use f-string here and above? Also seems like a perfect use of the `=` formatting:

```python
ttl = 100
limit = 50
print(f"{ttl=} is greater than {limit}")
ttl=100 is greater than 50
```

**Review by pb8o — APPROVED** [review]

File: src/mmds/src/token.rs (hunk 1 above)
Comment (roypat) [reply src/mmds/src/token.rs]: Ah, this is a rust test, not a python one, and it seems `=` formatting is not a thing in rust :(

Comment (pb8o) [reply src/mmds/src/token.rs]: Doh! Sorry, somehow I assumed this was Python 🤦

**Review by pb8o — APPROVED** [review]

File: src/mmds/src/token.rs [hunk 2]
```diff
@@ -378,12 +378,15 @@ mod tests {
         // We allow a deviation of 20ms to account for the gap
         // between the two calls to `get_time_ms()`.
         let deviation = 20;
-        assert!(ttl >= MILLISECONDS_PER_SECOND - deviation && ttl <= MILLISECONDS_PER_SECOND);
+        assert!(
+            ttl >= MILLISECONDS_PER_SECOND - deviation && ttl <= MILLISECONDS_PER_SECOND,
+            "ttl={ttl} not within +-{deviation} of {MILLISECONDS_PER_SECOND}",
```
Comment (zulinx86) [inline src/mmds/src/token.rs:383]: This condition evaluates the `ttl` between `MILLISECONDS_PER_SECOND - deviation` and `MILLISECONDS_PER_SECOND`, not between `MILLISECONDS_PER_SECOND - deviation` and `MILLISECONDS_PER_SECOND + deviation`. So the message `+-{deviation}` looks not be appropriate or the condition itself might not be proper.

**Review by zulinx86 — CHANGES_REQUESTED** [review]

File: src/mmds/src/token.rs (hunk 2 above)
Comment (roypat) [reply src/mmds/src/token.rs:383]: You are right, good catch! `ttl` can only every be larger than 1 second, not smaller, so the assert allows deviation in exactly the wrong direction. I would assume that this is indeed the reason we were observing the spurious failures.

**Review by zulinx86 — APPROVED** [review]

**Review by pb8o — APPROVED** [review]

**Pull request merged**